src/components/AccountDrawer/PrefetchBalancesWrapper.tsx

```diff
@@ -27,12 +27,12 @@ function useHasUpdatedTx() {
   const previousPendingTxs = usePrevious(pendingTxs)
 
   return useMemo(() => {
-    if (!previousPendingTxs) return false
+    if (!previousPendingTxs || !account) return false
     return Object.values(currentChainTxs).some(
-      (tx) => !isTxPending(tx) && wasPending(previousPendingTxs, tx),
+      (tx) => tx.from === account && !isTxPending(tx) && wasPending(previousPendingTxs, tx),
```

Comment:
what about `tx.to`?

Reply:
![image](https://github.com/Uniswap/interface/assets/5773490/8c4a4d4d-ef06-421c-a8cf-a872b9fecdd4)
I think you should be able to add `tx.to` to the transaction adder from the ethers `TransactionResponse`

Reply:
@tinaszheng I think that is out of scope for this PR as it would mean switching chain would remove any indication of a pending transaction, which might be desired but seems like a product choice beyond this bug fix.

Reply:
@JFrankfurt transactions on interface that are put into the redux store will still only be "from" and not "to" the current account

Reply:
we don't have any use for a "to" field currently

Reply:
yeah don't think we need to do `tx.to` here - that's mostly for contract addresses in the context of swap

Reply:
@cartcrom it's not based on chain switching right? it's based on account switching, and removing pending tx based on account seems logical to me. you can also define a separate hook that just filters out based on active address and use that exclusively for this flow

Reply:
ahh my bad I had totally conflated the two when I read your comment. yes will update or add new hook

Reply:
talked to you in office abt adding a todo comment instead 🫡